File: C/Stack/Array_Stack/STACK.c
#include "STACK.h"

void Stack_init(STACK *stack)
{
    stack->capacity = 2;
    stack->top = -1;
    stack->size = 0;
    stack->array = (int *)malloc(sizeof(int) * (stack->capacity));
}

int Stack_Size(STACK *stack)
{
    return stack->size;
}

int isEmpty(STACK *stack)
{
    return Stack_Size(stack) == 0;
}

void push(STACK *stack, int value)
{
    if (!stack)
        return;

    if (stack->size == stack->capacity)
    {
        stack->array = (int *)realloc(stack->array, sizeof(int) * (stack->capacity + 2));
        stack->capacity += 2;
    }

    stack->size++;
    stack->top++;

    stack->array[stack->top] = value;
}

int pop(STACK *stack)
{
    if (stack->top > -1)
    {
        int val = stack->array[stack->top];
        stack->top--;
        stack->size--;
        return val;
    }
    else
    {
        printf("Stack is empty\n");
        return -1;
    }
}

int top(STACK *stack)
{
    if (stack->top == -1)
    {
        return -1;
    }

    return stack->array[stack->top];
}

File: Java/Stack/Array_Stack/Stack.java
package Java.Stack.Array_Stack;

public class Stack {

    private int Top;
    private int length;
    private int capacity;
    private int[] array;

    Stack(int size) {
        this.Top = -1;
        this.length = 0;
        this.capacity = size;
        this.array = new int[size];
    }

    Stack() {
        this.Top = -1;
        this.length = 0;
        this.capacity = 2;
        this.array = new int[2];
    }

    private void resize() {
        int[] temp = array;
        capacity += 2;
        array = new int[capacity];

        for (int i = 0; i < length; i++) {
            array[i] = temp[i];
        }
    }

    public int top() {
        return array[Top];
    }

    public int size() {
        return length;
    }

    public boolean empty() {
        return length == 0;
    }

    public void push(int value) {
        if (length == capacity) {
            resize();
        }
        array[++Top] = value;
        length++;
    }

    public void pop() {
        if (!empty()) {
            Top--;
            length--;
        }
    }
}